connectors/kafka/src/main/java/forklift/controller/KafkaController.java

```diff
@@ -145,32 +114,20 @@ public void start() {
      * @throws InterruptedException if the thread is interrupted
      */
     public void stop(long timeout, TimeUnit timeUnit) throws InterruptedException {
-        kafkaConsumer.wakeup();
-        executor.shutdownNow();
-
         running = false;
+        kafkaConsumer.wakeup();
+       // executor.shutdownNow();
+        executor.shutdown();
```

Comment:
Yeah, whoops.

Hmm, maybe the designers wanted to be able to be sure that when `awaitTermination` is called that no new threads are going to be executed (because that makes the check more difficult than just a bunch of `Thread.join` calls). Seems good then.